src/server/main_service.cc

```diff
@@ -656,9 +658,13 @@ Transaction::MultiMode DeduceExecMode(ExecScriptUse state,
   if (!transactional && exec_info.watched_keys.empty())
     return Transaction::NOT_DETERMINED;
 
+  if (contains_admin_cmd) {
+    multi_mode = Transaction::NON_ATOMIC;
```

Comment:
Wouldn't this break multi/exec that contain admin commands but should run atomically ?

Reply:
As discussed f2f this does brakes atomicity but we do not exepect admin commands with transactional commands.
Having said that we can change this back once we refactor replication code and remove the mutex non cpu ops